bus_analyzer/bus_analyzer/extract.py

```diff
@@ -9,29 +9,101 @@
 from common.segments import extract_frame, parse_segment_path
 
 
-# DB2024 1080p buscam
-# bounding box (left x, top y, right x, bottom y) of the area the odometer can be in
-AREA_COORDS = {
-	"odo": (1053, 857, 1170, 930),
-	"clock": (1498, 852, 1590, 910)
+colour_profiles = {
+	'DBfH_2025':{
+		'sky_colours': {
+			'score': (113, 118, 114),
+			'day': (85, 203, 200),
+			'dusk': (217, 150, 181),
+			'night': (0, 0, 0),
+			'dawn': (36, 38, 117),
+		},
+		'dash_colours': {
+			'score': (171, 173, 143),
+			'day': (144, 0, 0),
+			'dusk': (115, 0, 0),
+			'night': (41, 0, 0),
+			'dawn': (78, 0, 0),
+		},
+	},
+	'DBfH_2024':{
+		'sky_colours': {
+			'score': (119, 119, 119),
+			'day': (82, 218, 217),
+			'dusk': (217, 150, 181),
+			'night': (0, 0, 0),
+			'dawn': (36, 38, 117),
+		},
+		'dash_colours': {
+			'score': (181, 181, 150),
+			'day': (146, 0, 1),
+			'dusk': (115, 0, 0),
+			'night': (41, 0, 0),
+			'dawn': (78, 0, 0),
+		},
+	},
+	# for reference, before we started deteching the score screen
+	# or using the colour of the dash
+	'DBfH_2023': {
+		'sky_colour': {
+			'day': (89, 236, 239),
+			'dusk': (199, 162, 205),
+			'night': (1, 1, 1),
+			'dawn': (51, 59, 142),			
+		}
+	},
 }
+
+# 'area_coords'
+# bounding box (left x, top y, right x, bottom y) of the area the odometer can be in
+# 'digit_x_coords'
 # starting x coord of each digit within the odo box
-DIGIT_X_COORDS = {
-	"odo": [0, 22, 44, 66, 96],
-	"clock": [0, 22, 53, 75],
-}
+# 'digit_bases'
 # value of each digit
-DIGIT_BASES = {
-	"odo": [1000, 100, 10, 1, 0.1],
-	"clock": [600, 60, 10, 1],
-}
-
-DIGIT_WIDTH = 17
+# 'digit_height'
 # Most digits we only care about the actual character height
-DIGIT_HEIGHT = 24
+# 'last_digit_height'
 # But last digit we want the full white-background area as we want to try to match
 # based on position also.
-LAST_DIGIT_HEIGHT = 39
+location_profiles = {
+	'DBfH_2024': {
+		'area_coords': {
+			'odo': (1053, 857, 1170, 930),
+			'clock': (1498, 852, 1590, 910)
+		},
+		'digit_x_coords': {
+			'odo': (0, 22, 44, 66, 96),
+			'clock': (0, 22, 53, 75)
+		},
+		'digit_bases': {
+			'odo': (1000, 100, 10, 1, 0.1),
+			'clock': (600, 60, 10, 1)
+		}, 
+		'digit_width': 17,
+		'digit_height': 24,
+        'last_digit_height': 39,
+		'sky_pixel': (1614, 192),
+		'dash_pixel': (945, 864),
+	},
+	'DBfH_2023': {
+		'area_coords': {
+			'odo': (1121, 820, 1270, 897),
+			'clock': (1685, 819, 1804, 877)
+		},
+		'digit_x_coords': {
+			'odo': (0, 28, 56, 84, 123),
+			'clock': (0, 27, 66, 93),
+		},
+		'digit_width': 26,
+		'digit_height': 26,
+        'last_digit_height': 38,
+		'sky_pixel': (177, 255),
+	},
+}
+
+profiles = {}
+profiles['DBfH_2025'] = colour_profiles['DBfH_2025'] | location_profiles['DBfH_2024']
+profiles['DBfH_2024'] = colour_profiles['DBfH_2024'] | location_profiles['DBfH_2024']
```

Comment:
nitpick but let's avoid needing to hard-code each key again:
```suggestion
profiles = {name: colour_profiles[name] | location_profiles[name] for name in set(color_profiles) | set(location_profiles)}
```

Reply:
In the end I just kept your version